packages/test-app/tests/integration/components/validation-error-test.js

```diff
@@ -7,27 +7,43 @@ module('Integration | Component | validation-error', function (hooks) {
   setupRenderingTest(hooks);
 
   test('it renders with no errors', async function (assert) {
-    await render(hbs`<ValidationError @errors={{(array)}} />`);
+    assert.expect(2);
+
+    const propertyName = 'foo';
+
+    this.set('validatable', {
+      async getErrorsFor(prop) {
+        assert.strictEqual(prop, propertyName);
+        return [];
+      },
+    });
+    this.set('propertyName', propertyName);
+
+    await render(
+      hbs`<ValidationError @validatable={{this.validatable}} @property={{this.propertyName}} />`,
+    );
 
     assert.dom(this.element).hasText('');
   });
 
   test('it renders with errors', async function (assert) {
-    await render(hbs`<ValidationError @errors={{array 'test 1' 'test 2'}} />`);
+    assert.expect(2);
 
-    assert.dom(this.element).hasText('test 1 test 2');
-  });
-
-  test('it responds to changes', async function (assert) {
```

Comment:
I think this test got left out and I'm worried about how this implementation will respond to new errors. Does it know to call the promise again, or will it never update?

Reply:
I'm actually glad you took a look at this PR, as I'm not the expert on this functionality :)

Reply:
we have plenty of indirect test coverage on state changes of the validated attribute, via integration tests for components that use the `<ValidationError>` component.

transforming/keeping this test-case would require a dummy object that's responsive. that seems like a lot of work for very little direct gain here.

Reply:
Summary of conversation: this isn't easily testable, but there is plenty of coverage for it being correctly reactive.